apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.html.ts

```diff
@@ -0,0 +1,563 @@
+        groupB,three@example.com<br />
+        groupB,four@example.com</code
+      >
+      <!-- Closing code tag not on its own line to improve copy/paste formatting -->
+      </div>
+      <div class="modal-body">
+        <p>
+          The <code>groupname</code> column should be a unique identifier for each group. To
+          change the grouping, link uids to the groupname.
+        </p>
+        <form name="upload-assessment-group-form" method="POST" enctype="multipart/form-data">
+          <div class="form-group">
+            <div class="custom-file">
+              <input
+                type="file"
+                accept=".csv"
+                name="file"
+                class="custom-file-input"
+                id="uploadAssessmentGroupsFileInput"
+              />
+              <label class="custom-file-label" for="uploadAssessmentGroupsFileInput">
+                Choose CSV file
+              </label>
+            </div>
+          </div>
+          <div class="d-flex justify-content-end">
+            <div class="form-group mb-0">
+              <input type="hidden" name="__action" value="upload_assessment_groups" />
+              <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+              <button type="button" class="btn btn-secondary" data-dismiss="modal">
+                Cancel
+              </button>
+              <button type="submit" class="btn btn-primary">Upload</button>
+            </div>
+          </div>
+        </form>
+      </div>
+    `,
+  });
+}
+
+function AutoAssessmentGroupsModal({
+  groupMin,
+  groupMax,
+  csrfToken,
+}: {
+  groupMin: number;
+  groupMax: number;
+  csrfToken: string;
+}) {
+  return Modal({
+    id: 'autoAssessmentGroupsModal',
+    title: 'Auto new group setting',
+    body: html`
+      <form name="auto-assessment-group-form" method="POST" enctype="multipart/form-data">
+        <div class="form-group">
+          <label for="formMin">Min number of members in a group</label>
+          <input
+            type="text"
+            value="${groupMin}"
+            class="form-control"
+            id="formMin"
+            name="min_group_size"
+          />
+        </div>
+        <div class="form-group">
+          <label for="formMax">Max number of members in a group</label>
+          <input
+            type="text"
+            value="${groupMax}"
+            class="form-control"
+            id="formMax"
+            name="max_group_size"
+          />
+        </div>
+        <div class="d-flex justify-content-end">
+          <div class="form-group mb-0">
+            <input type="hidden" name="__action" value="auto_assessment_groups" />
+            <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+            <button type="button" class="btn btn-secondary" data-dismiss="modal">Cancel</button>
+            <button type="submit" class="btn btn-primary">Group</button>
+          </div>
+        </div>
```

Comment:
I know you're just copying this along from existing code, but this isn't the conventional way to display action buttons on a modal. You should use `footer` here like the other two modals in this file do.

Reply:
Moved these buttons to the `footer` in [dc8a69d](https://github.com/PrairieLearn/PrairieLearn/pull/9550/commits/dc8a69d5bba27345198abea501037c2deb732050).